test/test.cc

```diff
@@ -692,7 +692,7 @@ int run(int argc, char** argv)
 
         // After parsing parameters, call test routine again (with run=false)
         // to mark any new fields as used (e.g., timers).
-        test_routine( params, false );
+        //test_routine( params, false );
```

Comment:
I'm assuming this needs to be uncommented once this is working.

Reply:
For the timers, I called the `test_routine` twice with `run = false`, once before parsing parameters, and a new call after parsing parameters, since we don't know what timer columns to include until after parsing parameters. But this screws up Cholesky for setting the matrix type. By default, it is set to `rand_dominant`, before parsing parameters. Then the user can override it with parameters. But now, it is reset to `rand_dominant` *after* parsing parameters, overriding the user's choice. Ugh.

Reply:
Would it make sense to add a `get_default` to the Params classes and do something like:
```C++
if (!run && params.matrix.kind.get_default() != "rand_dominant") {
  params.matrix.kind.set_default("rand_dominant");
}
```
Or maybe add a accessor for `ParamBase::is_default_` and check that?

Reply:
Fixed in https://github.com/icl-utk-edu/testsweeper/pull/19